idea-plugin/src/main/kotlin/arrow/meta/ide/testing/env/types/HeavyTestSyntax.kt

```diff
@@ -1,3 +1,100 @@
 
-object HeavyTestSyntax
\ No newline at end of file
+import arrow.meta.ide.phases.config.buildFolders
+import arrow.meta.ide.testing.Source
+import arrow.meta.ide.testing.env.types.HeavyTestSyntax.plus
+import arrow.meta.plugin.testing.CompilationData
+import arrow.meta.plugin.testing.CompilationResult
+import arrow.meta.plugin.testing.CompilerPlugin
+import arrow.meta.plugin.testing.Config
+import arrow.meta.plugin.testing.ConfigSyntax
+import arrow.meta.plugin.testing.Dependency
+import arrow.meta.plugin.testing.compilationData
+import arrow.meta.plugin.testing.compile
+import com.intellij.openapi.application.WriteAction
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ModuleRootManager
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.testFramework.PsiTestUtil
+import com.intellij.testFramework.fixtures.CodeInsightTestFixture
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+data class HeavyTestSetUp(
+  val buildDir: VirtualFile,
+  val srcDir: VirtualFile,
+  val ktFile: KtFile,
+  val traversable: List<PsiElement>,
+  val outDirTarget: VirtualFile
+)
+
+val compilerPlugin = CompilerPlugin("Arrow Meta", listOf(Dependency("compiler-plugin")))
+val arrowAnnotations = { str: String -> Dependency("arrow-annotations:$str") }
+val arrowCoreData = { str: String -> Dependency("arrow-core-data:$str") }
+
+val defaultConfig: (String) -> List<Config>
+  get() = { str -> HeavyTestSyntax.addCompilerPlugins(compilerPlugin) + HeavyTestSyntax.addDependencies(arrowAnnotations(str), arrowCoreData(str)) }
+
+object HeavyTestSyntax : CommonTestSyntax, ConfigSyntax {
+  override val emptyConfig: Config
+    get() = Config.Empty
+
+  fun Source.compile(config: List<Config>): CompilationResult =
+    compile(config.compilationData(CompilationData(source = listOf(this.trimMargin()))))
+
+  val CompilationResult.outDirFile: VirtualFile?
+    get() = TODO("::outputDirectory")
+
+  fun Source.addMetaDataToBuild(config: List<Config>, buildDir: VirtualFile, myFixture: CodeInsightTestFixture): VirtualFile? =
+    compile(config).outDirFile?.let { myFixture.copyDirectoryToProject(it.path, buildDir.path) }
+
+  fun Source.toFile(name: String = "Source.kt", dir: VirtualFile): VirtualFile? =
+    WriteAction.computeAndWait<VirtualFile, Throwable> {
+      val sourceFile = dir.createChildData(this, name)
+      sourceFile.setBinaryContent(toByteArray())
+      sourceFile
+    }
+
+  fun Source.copyToDir(name: String = "Source.kt", dir: VirtualFile, myFixture: CodeInsightTestFixture): KtFile? =
+    toFile(name, dir)?.let { file: VirtualFile ->
+      myFixture.configureFromExistingVirtualFile(file)
+      myFixture.file.fileType.safeAs()
+    }
+
+  /**
+   * @param receiver is the root file
+   */
+  fun VirtualFile.addDir(name: String, module: Module): VirtualFile? =
```

Comment:
At the moment this is always adding an excluded folder (see 2 lines below). I'll rename this to `addExcludedDir` and add method `addSourceDir`